src/worker/core/GameSim.basketball/index.ts

```diff
@@ -2033,47 +2015,46 @@ class GameSim extends GameSimBase {
 			if (r1 > r2 && r1 > r3) {
 				// Two point jumper
 				type = "midRange";
+				fgaLogType = "fgaMidRange";
 				probMissAndFoul = 0.07;
 				probMake =
 					this.team[this.o].player[p].compositeRating.shootingMidRange * 0.32 +
 					0.42;
 				probAndOne = 0.05;
-				this.playByPlay.logEvent({
-					type: "fgaMidRange",
-					t: this.o,
-					pid: this.team[this.o].player[p].id,
-					clock: this.t,
-				});
 			} else if (r2 > r3) {
 				// Dunk, fast break or half court
 				type = "atRim";
+				fgaLogType = "fgaAtRim";
 				probMissAndFoul = 0.37;
 				probMake =
 					this.team[this.o].player[p].compositeRating.shootingAtRim * 0.41 +
 					0.54;
 				probAndOne = 0.25;
-				this.playByPlay.logEvent({
-					type: "fgaAtRim",
-					t: this.o,
-					pid: this.team[this.o].player[p].id,
-					clock: this.t,
-				});
 			} else {
 				// Post up
+				fgaLogType = "fgaLowPost";
 				type = "lowPost";
 				probMissAndFoul = 0.33;
 				probMake =
 					this.team[this.o].player[p].compositeRating.shootingLowPost * 0.32 +
 					0.34;
 				probAndOne = 0.15;
-				this.playByPlay.logEvent({
-					type: "fgaLowPost",
-					t: this.o,
-					pid: this.team[this.o].player[p].id,
-					clock: this.t,
-				});
 			}
-
+			this.playByPlay.logEvent({
+				type: fgaLogType,
+				t: this.o,
+				pid: this.team[this.o].player[p].id,
+				clock: this.t,
+				desperation:
+					(fgaLogType as string) === "fgaTp" ||
+					(fgaLogType as string) === "fgaTpFake"
+						? rushed && forceThreePointer
+						: undefined,
+				pidPass:
+					(fgaLogType as string) === "fgaTipIn" && passer
```

Comment:
Casting like `as string` should be avoided if possible since often it results in bugs. Is it really needed here?